solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetAccessStatusFilter.java

```diff
@@ -0,0 +1,105 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.data.ingest.solr.filter;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.unc.lib.dl.acl.fcrepo4.InheritedAclFactory;
+import edu.unc.lib.dl.acl.fcrepo4.ObjectAclFactory;
+import edu.unc.lib.dl.acl.service.PatronAccess;
+import edu.unc.lib.dl.acl.util.AccessPrincipalConstants;
+import edu.unc.lib.dl.data.ingest.solr.exception.IndexingException;
+import edu.unc.lib.dl.data.ingest.solr.indexing.DocumentIndexingPackage;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.search.solr.util.FacetConstants;
+/**
+ * Sets access-related status tags
+ *
+ * @author harring
+ *
+ */
+public class SetAccessStatusFilter implements IndexDocumentFilter {
+    private static final Logger log = LoggerFactory.getLogger(SetAccessStatusFilter.class);
+    private InheritedAclFactory inheritedAclFactory;
+    private ObjectAclFactory objAclFactory;
+
+    @Override
+    public void filter(DocumentIndexingPackage dip) throws IndexingException {
+        log.debug("Performing set access status filter on {}", dip.getPid());
+        dip.getDocument().setStatus(determineAccessStatus(dip));
+    }
+
+    /**
+     * Sets inherited acl factory
+     *
+     * @param aclFactory an inherited acl factory
+     */
+    public void setInheritedAclFactory(InheritedAclFactory iaf) {
+        this.inheritedAclFactory = iaf;
+    }
+
+    /**
+     * Sets non-inherited acl factory
+     *
+     * @param aclFactory an inherited acl factory
+     */
+    public void setObjectAclFactory(ObjectAclFactory oaf) {
+        this.objAclFactory = oaf;
+    }
+
+    private List<String> determineAccessStatus(DocumentIndexingPackage dip)
+            throws IndexingException {
+
+        PID pid = dip.getPid();
+        List<String> status = new ArrayList<>();
+
+        PatronAccess inheritedAccess = inheritedAclFactory.getPatronAccess(pid);
+        PatronAccess objAccess = objAclFactory.getPatronAccess(pid);
+
+        if (inheritedAclFactory.isMarkedForDeletion(pid)) {
+            status.add(FacetConstants.MARKED_FOR_DELETION);
+        }
+
+        Date objEmbargo = objAclFactory.getEmbargoUntil(pid);
+        Date parentEmbargo = inheritedAclFactory.getEmbargoUntil(pid);
+        if (parentEmbargo != null && objEmbargo != null) {
+            status.add(FacetConstants.EMBARGOED_PARENT);
+            status.add(FacetConstants.EMBARGOED);
+        } else if (parentEmbargo != null) {
+            status.add(FacetConstants.EMBARGOED_PARENT);
+        } else if (objEmbargo != null) {
+                status.add(FacetConstants.EMBARGOED);
+        }
+
+        if (inheritedAccess.equals(PatronAccess.none)) {
+            status.add(FacetConstants.STAFF_ONLY_ACCESS);
```

Comment:
One of the tests was weirding me out (testParentHasStaffOnlyAccess).  Then I realized it was because STAFF_ONLY_ACCESS (as compared to PARENT_HAS_STAFF_ONLY_ACCESS) should only be added if the object itself is marked as PatronAccess.none.  The inherited value shouldn't matter here.